### PR TITLE
fix: remove unused dependency

### PR DESCRIPTION
**Pull request by SandZn**
Hi,

I found that one dependency (inherits) is not used in your package, according to your test. So I modified the package.json file to exclude the useless dependency. Would you consider removing useless dependencies from your package, so that developers do not need to install them when they use your package?

### CODE REVIEW

**Review by ralphtheninja — APPROVED** [review]

**Comment by vweevers** [comment]
It's used here: https://github.com/Level/subleveldown/blob/58db7bcbce3f77a60360e7669082bd2a45fc520f/leveldown.js#L3

**Comment by SandZn** [comment]
Hi Vincent,

Thanks for your fast response 😊.

I am doing research on finding unnecessary dependencies in nodejs packages. I understand that there are statements in files showing the dependency is "used"; for example, they are required in some files. I would like to know if the dependency is really necessary, why the package still manages to pass the tests after we remove it from package.json and do not install it at all. 

How is it necessary for your package? If it is, why do the tests not test them? If not, why do we still need them?

**Comment by vweevers** [comment]
> How is it necessary for your package?

It does exactly what the description of `inherits` says.

> Why do the tests not test them?

Because it's out of scope and covered by the tests of `inherits`.

> why [does] the package still manage to pass the tests after we remove `inherits` from `package.json`?

Running `npm ls inherits` will give you the answer, which is that several dependencies also depend on `inherits`, meaning that npm will install it regardless of whether `subleveldown` explicitly declares it in its `package.json`.

**Comment by ralphtheninja** [comment]
@vweevers Sorry about the approval btw.

**Comment by vweevers** [comment]
No worries :)

**Comment by SandZn** [comment]
> Running npm ls inherits will give you the answer.

Yes. It could be "used" not only by `subleveldown`, but also by transitive dependencies. But we can only handle the direct dependencies. For those transitive dependencies, we don't know and can't really believe they are harmless.

> several dependencies also depend on inherits, meaning that npm will install it regardless of whether subleveldown explicitly declares it in its package.json

Exactly. Then to exclude `inherits` (if it's really not used) from the whole dependency tree of `subleveldown`, it's not enough to just modify package.json. We need to find another approach to cut down `inherits` (if it's unused) from the transitive dependencies.

**Comment by vweevers** [comment]
@SandZn What is your remaining goal? We've established that `inherits` is used, and explained the flaw in your method (for determining whether it's used). BTW and FYI: we have a continuous [test](https://github.com/Level/subleveldown/blob/58db7bcbce3f77a60360e7669082bd2a45fc520f/package.json#L13) to warn us about unused dependencies.

**Comment by SandZn** [comment]
Hi @vweevers, no need to worry and do not misunderstand:), I do not insist on removing `inherits` from `subleveldown`. I'm 
 discussing my ideas and would like to know what developers think. 

By using my approach, I find that `inherits` is not necessary, that is because after I remove `inherits` from package.json, I run `npm install` and `npm run test`,  `subleveldown` still passes all tests. 

If `inherits` is necessary for `subleveldown`, why do the tests not check **`subleveldown` is using `inherits`**? (This is my remaining confusion.) I'm not asking why `subleveldown` is not testing `inherits`. I'm asking why `subleveldown` is not testing the use of `inherits`.

There are cases in some projects showing that even if they install a dependency, they actually do not need it.

> for determining whether it's used

So in my opinion, the `npm list` result can not convince me that a dependency is really necessary for a package. 

I admit that some tools also check unused dependencies, like [depcheck](https://github.com/depcheck/depcheck) and [dependency-check](https://github.com/dependency-check-team/dependency-check). They use different approaches than my approach.

**Comment by ralphtheninja** [comment]
> If `inherits` is necessary for `subleveldown`, why do the tests not check **`subleveldown` is using `inherits`**? (This is my remaining confusion.) I'm not asking why `subleveldown` is not testing `inherits`. I'm asking why `subleveldown` is not testing the use of `inherits`.

This is due to how `node.js` works when you require a dependency and there's no way to get around that without modifying how `node.js` works. If you remove `inherits` from `package.json` the tests still pass because there's another dependency also using it, so you could make the case that `inherits` is also being used _indirectly_ by `subleveldown` but then you rely on other dependencies to pull in dependencies for you, which is a rug waiting to be pulled.

In other words, your approach has a flaw in it. Removing a dependency and having the tests pass doesn't imply that the dependency in question (in this case `inherits`) isn't needed.
